### PR TITLE
Support non-default pulse settings

### PR DESCRIPTION
**Pull request by at-wat**
Set PULSE_SERVER and PULSE_COOKIE environment variables to specify
server and cookie location. If cookie file doesn't exist, try
anonymous auth.
This commit also adds support for TCP socket connection.

### CODE REVIEW

**Comment by at-wat** [comment]
ref: https://www.freedesktop.org/wiki/Software/PulseAudio/FAQ/#whatenvironmentvariablesdoespulseaudiocareabout

File: client.go [hunk 1]
```diff
@@ -21,7 +23,18 @@ type Client struct {
 }
 
 func NewClient() (*Client, error) {
-	conn, err := net.Dial("unix", fmt.Sprint("/run/user/", os.Getuid(), "/pulse/native"))
+	socketNetwork := "unix"
+	socketPath := fmt.Sprint("/run/user/", os.Getuid(), "/pulse/native")
+	if pathRaw, ok := os.LookupEnv("PULSE_SERVER"); ok {
+		path := strings.SplitN(pathRaw, ":", 2)
+		if len(path) != 2 {
```
Comment (at-wat) [inline client.go]: https://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/User/ServerStrings/
If the network type is not specified, it should be handled as unix socket.

Comment (at-wat) [reply client.go]: fixed by 9737046

**Comment by jfreymuth** [comment]
a ':' can also indicate a port number. Also, a server string can contain multiple addresses.

Here is my WIP implementation of server string parsing:

	type serverString struct {
		localname string
		protocol  string
		addr      string
	}

	func parseServerString(str string) []serverString {
		s := strings.Fields(str)
		result := make([]serverString, len(s))
		for i, s := range s {
			if s[0] == '{' {
				end := strings.IndexByte(s, '}')
				result[i].localname = s[1:end]
				s = s[end+1:]
			}
			switch {
			case s[0] == '/':
				result[i].protocol = "unix"
				result[i].addr = s
			case strings.HasPrefix(s, "unix:"):
				result[i].protocol = "unix"
				result[i].addr = s[5:]
			case strings.HasPrefix(s, "tcp6:"):
				result[i].protocol = "tcp6"
				result[i].addr = s[4:]
			case strings.HasPrefix(s, "tcp4:"):
				result[i].protocol = "tcp4"
				result[i].addr = s[4:]
			case strings.HasPrefix(s, "tcp:"):
				result[i].protocol = "tcp4"
				result[i].addr = s[3:]
			default:
				result[i].addr = s
			}
		}
		return result
	}

**Pull request merged**